src/osd/PrimaryLogPG.cc

```diff
@@ -11022,14 +11044,38 @@ uint64_t PrimaryLogPG::recover_backfill(
     add_object_context_to_pg_stat(obc, &stat);
     pending_backfill_updates[*i] = stat;
   }
-  for (unsigned i = 0; i < to_remove.size(); ++i) {
-    handle.reset_tp_timeout();
+  if (HAVE_FEATURE(get_min_upacting_features(), SERVER_LUMINOUS)) {
+    map<pg_shard_t,MOSDPGBackfillRemove*> reqs;
+    for (unsigned i = 0; i < to_remove.size(); ++i) {
+      handle.reset_tp_timeout();
+      pg_shard_t target = to_remove[i].get<2>();
+      MOSDPGBackfillRemove *m;
+      if (reqs.count(target)) {
+	m = reqs[target];
```

Comment:
I think we're considering this an anti-pattern now, and should use iterator find()?